(python3.4 + Ubuntu 14.04) specific fallout with long filenames

While building pkgs for neurodebian, 14.04 build failed with
```
======================================================================
ERROR: datalad.support.tests.test_annexrepo.test_files_split_exc(<class 'datalad.support.annexrepo.AnnexRepo'>,)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/build/datalad-0.11.2/.pybuild/pythonX.Y_3.4/build/datalad/tests/utils.py", line 615, in newfunc
    return t(*(arg + (filename,)), **kw)
  File "/build/datalad-0.11.2/.pybuild/pythonX.Y_3.4/build/datalad/support/tests/test_annexrepo.py", line 2512, in check_files_split_exc
    r.add(files)
  File "/build/datalad-0.11.2/.pybuild/pythonX.Y_3.4/build/datalad/support/gitrepo.py", line 316, in newfunc
    result = func(self, files_new, *args, **kwargs)
  File "/build/datalad-0.11.2/.pybuild/pythonX.Y_3.4/build/datalad/support/annexrepo.py", line 1475, in add
    git_options=git_options, annex_options=annex_options, update=update
  File "/build/datalad-0.11.2/.pybuild/pythonX.Y_3.4/build/datalad/support/annexrepo.py", line 1550, in add_
    for rec in to_be_added_recs
  File "/build/datalad-0.11.2/.pybuild/pythonX.Y_3.4/build/datalad/support/annexrepo.py", line 1549, in <dictcomp>
    rec['file']: self.get_file_size(rec['file'])
  File "/build/datalad-0.11.2/.pybuild/pythonX.Y_3.4/build/datalad/support/annexrepo.py", line 1520, in _get_to_be_added_recs
    files, git_options=_git_options, update=update):
  File "/build/datalad-0.11.2/.pybuild/pythonX.Y_3.4/build/datalad/support/gitrepo.py", line 1089, in add_
    to_options(update=update) + ['--verbose']
  File "/build/datalad-0.11.2/.pybuild/pythonX.Y_3.4/build/datalad/support/annexrepo.py", line 1261, in _git_custom_command
    return super(AnnexRepo, self)._git_custom_command(*args, **kwargs)
  File "/build/datalad-0.11.2/.pybuild/pythonX.Y_3.4/build/datalad/support/gitrepo.py", line 316, in newfunc
    result = func(self, files_new, *args, **kwargs)
  File "/build/datalad-0.11.2/.pybuild/pythonX.Y_3.4/build/datalad/support/gitrepo.py", line 1748, in _git_custom_command
    expect_fail=expect_fail)
  File "/build/datalad-0.11.2/.pybuild/pythonX.Y_3.4/build/datalad/support/gitrepo.py", line 1800, in _run_command_files_split
    *args, **kwargs)
  File "/build/datalad-0.11.2/.pybuild/pythonX.Y_3.4/build/datalad/cmd.py", line 683, in run
    cmd, env=self.get_git_environ_adjusted(env), *args, **kwargs)
  File "/build/datalad-0.11.2/.pybuild/pythonX.Y_3.4/build/datalad/cmd.py", line 496, in run
    stdin=stdin)
  File "/usr/lib/python3.4/subprocess.py", line 848, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.4/subprocess.py", line 1446, in _execute_child
    raise child_exception_type(errno_num, err_msg)
OSError: [Errno 7] Argument list too long

======================================================================
FAIL: datalad.support.tests.test_annexrepo.test_files_split_exc(<class 'datalad.support.gitrepo.GitRepo'>,)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/build/datalad-0.11.2/.pybuild/pythonX.Y_3.4/build/datalad/tests/utils.py", line 615, in newfunc
    return t(*(arg + (filename,)), **kw)
  File "/build/datalad-0.11.2/.pybuild/pythonX.Y_3.4/build/datalad/support/tests/test_annexrepo.py", line 2516, in check_files_split_exc
    assert_not_in('too long', str(ecm.exception))
AssertionError: 'too long' unexpectedly found in '[Errno 7] Argument list too long'

----------------------------------------------------------------------
Ran 887 tests in 1432.142s

FAILED (SKIP=99, errors=1, failures=1)
/tmp/datalad_temp_tree_test_paths_with_forward_slashesk2jsqatu
Versions: appdirs=1.3.0 boto=2.44.0 cmd:annex=7.20181121+git58-gbc4aa3f0e-1~ndall+1 cmd:git=2.19.2 cmd:system-git=1.9.1 cmd:system-ssh=6.6p1 exifread=2.1.2 git=2.1.8 gitdb=2.0.0 humanize=0.5.1 iso8601=0.1.10 keyring=3.5 msgpack=0.4.2 mutagen=1.38 patoolib=1.12 requests=2.2.1 six=1.9.0 tqdm=4.11.2 wrapt=1.9.0
```
after passing the tests for python 2.  Also it passed with python 3.4 in Debian 80. Versions of 3.4 are a bit different
```
neurodebian@smaug ~/deb/builds/datalad/0.11.2-1 % grep -A1 'Selecting.*libpython3.4-minimal' datalad_0.11.2-1*+1_amd64.build 
datalad_0.11.2-1~nd14.04+1_amd64.build:Selecting previously unselected package libpython3.4-minimal:amd64.
datalad_0.11.2-1~nd14.04+1_amd64.build-Preparing to unpack .../libpython3.4-minimal_3.4.0-2ubuntu1_amd64.deb ...
--
datalad_0.11.2-1~nd80+1_amd64.build:Selecting previously unselected package libpython3.4-minimal:amd64.
datalad_0.11.2-1~nd80+1_amd64.build-Preparing to unpack .../libpython3.4-minimal_3.4.2-1_amd64.deb ...
```
I am rerunning 14.04 build to see if it is reproducible.  If not -- means that there is more to fixup to guarantee robustness :-/  If reproducible - I guess we should blacklist/skip with 3.4 < 3.4.2

--- a/datalad/utils.py
+++ b/datalad/utils.py
@@ -89,6 +89,10 @@ CMD_MAX_ARG_HARDCODED = 2097152 if on_linux else 262144 if on_osx else 32767
 try:
     CMD_MAX_ARG = os.sysconf('SC_ARG_MAX')
     assert CMD_MAX_ARG > 0
+    if sys.version_info[:2] == (3, 4):
+        # workaround for some kind of a bug which comes up with python 3.4
+        # see https://github.com/datalad/datalad/issues/3150
+        CMD_MAX_ARG = min(CMD_MAX_ARG, CMD_MAX_ARG_HARDCODED)
 except Exception as exc:
     # ATM (20181005) SC_ARG_MAX available only on POSIX systems
     # so exception would be thrown e.g. on Windows, or

